lte/gateway/c/oai/oai_mme/CMakeLists.txt

```diff
@@ -62,3 +62,19 @@ endif ( NOT EMBEDDED_SGW )
 if ( NOT S6A_OVER_GRPC )
     target_link_libraries(mme fdproto fdcore)
 endif ( NOT S6A_OVER_GRPC )
+
+if ( NOT EMBEDDED_SGW AND NOT S6A_OVER_GRPC )
+  if (EXISTS "/etc/os-release")
+    set(os_file_path "/etc/os-release")
+    file(STRINGS "${os_file_path}" data_list REGEX "^(ID|VERSION_ID|VERSION_CODENAME)=")
+    foreach(_var ${data_list})
+      if("${_var}" MATCHES "^(ID)=(.*)$")
+        set(os_name "${CMAKE_MATCH_2}")
+      endif()
+    endforeach()
+    if ("${os_name}" STREQUAL "\"rhel\"")
+      target_link_libraries(mme
+        gpr yaml z cares boost_program_options boost_filesystem boost_system boost_regex double-conversion address_sorting gflags unwind iberty dl)
+    endif()
+  endif()
+endif ( NOT EMBEDDED_SGW AND NOT S6A_OVER_GRPC )
```

Comment:
you mean here that all other non MME S11/S6a deployments won't build on RHEL. 
It is the case now I imagine.
We could detect the OS as you do and not bother with EMBEDDED_SGW, S6A_OVER_GRPC ?

Reply:
I wanted to be the less intrusive possible.

Currently adding these "missing" libraries is only needed for that variant.